crates/svm-ffi/src/value.rs

```diff
@@ -31,25 +31,69 @@ use crate::svm_byte_array;
 /// |          | (1 byte)  |   bytes)  |        |   (1 byte)  |   bytes)   |
 /// +----------+--------------------------------+-------------+------------+
 ///
+
+/// Allocates a raw WASM values slice.
+/// The types for each item are determined by the input layout `types`.
+/// Each WASM value is assigned with zero.
+///
+/// # Example
+///
+/// ```rust
+/// use std::convert::TryFrom;
+///
+/// use svm_types::{WasmType, WasmValue};
+/// use svm_ffi::alloc_wasm_values;
+///
+/// let types = vec![WasmType::I64, WasmType::I32, WasmType::I64];
+/// let values = alloc_wasm_values(&types);
+///
+/// let values = Vec::<WasmValue>::try_from(&values).unwrap();
+/// assert_eq!(values, vec![WasmValue::I64(0), WasmValue::I32(0), WasmValue::I64(0)]);
+/// ```
+///
+pub fn alloc_wasm_values(types: &[WasmType]) -> svm_byte_array {
+    let nvalues = types.len();
+
+    assert!(nvalues <= std::u8::MAX as usize);
+
+    let capacity = 1 + nvalues * 9;
+
+    let mut bytes = Vec::with_capacity(capacity);
+
+    bytes.write_u8(nvalues as u8).unwrap();
+
+    for ty in types {
```

Comment:
I'd rather have a simpler version of this function, and have it to merely allocate the needed buffer size.
But you can leave as-is if you prefer.

Reply:
I've added a comment explaining why I prefer that we allocate the worst case.
Since memory allocation is a costly operation we should strive to minimize the number of its usages.

The savings of a few bytes is negligible in this case.

Reply:
That was my comment in https://github.com/spacemeshos/svm/pull/187#discussion_r528360072. In this comment I'm not referring to that, but to the filling of the allocated bytes (with `nvalues`, and `ty` for each value). Although the extern client must comply with the signature, I think it would be better for this function to just allocate the buffer.

Reply:
done